core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -296,10 +302,23 @@ public void onNetworkStateChanged() {
     }
   }
 
+  public void reinitialize(final MongoClient localClient) {
+    syncLock.lock();
+    try {
+      this.instanceChangeStreamListener.stop();
+      this.stop();
+      this.localClient = localClient;
+      initialize();
```

Comment:
I think the fact that this is going to be synchronous is a performance issue, but I imagine you will be fixing this in STITCH-2281?

Reply:
Yes, this can be fixed in 2281. The fact that our initialization is synchronous currently is already a performance issue, so I'm not worrying about it for this PR.